include/h2o.h

```diff
@@ -1088,12 +1092,13 @@ ssize_t h2o_find_header_by_str(const h2o_headers_t *headers, const char *name, s
 /**
  * adds a header to list
  */
-void h2o_add_header(h2o_mem_pool_t *pool, h2o_headers_t *headers, const h2o_token_t *token, const char *value, size_t value_len);
+h2o_header_t *h2o_add_header(h2o_mem_pool_t *pool, h2o_headers_t *headers, const h2o_token_t *token, const char *value,
+                             size_t value_len);
```

Comment:
How about adding `orig_hname` as an argument instead of returning `h2o_header_t *` so that you could set the value in the caller?

The same goes for `h2o_add_header_by_str` as well.